php-frontend/src/main/java/org/sonar/php/symbols/ClassSymbol.java

```diff
@@ -49,4 +52,6 @@ enum Kind {
     ABSTRACT,
     INTERFACE
   }
+
+  Trilean hasTestMethod();
```

Comment:
IMO this is a bit too much to check for test cases. It can also be okay to loop over the methods in a SmybolUtils class and not in the ClassSymbol itself. I would keep the symbol clean and without logic as long as possible.